apps/sms/views/conversation/js/conversation.js

```diff
@@ -2014,13 +2036,25 @@ var ConversationView = {
     ).then(() => {
       WaitingScreen.show();
 
-      var messageIdsToDelete = this.selectionHandler.selectedList.map(
-        (item) => +item
-      );
-
-      return this.deleteMessages(messageIdsToDelete);
+      if(ConversationView.inEditMode) {
+        messageIdsToDelete = this.selectionHandler.selectedList.map(
+          (item) => +item
+        );
+        return onAllThreadMessagesRetrieved(messageIdsToDelete);
+      } else {
+        console.log('test1');
+        MessageManager.getMessages({
+        // Filter and request all messages with this threadId
+          filter: { threadId: ConversationView.activeThread.id },
+          each: onThreadMessageRetrieved,
+          end: onAllThreadMessagesRetrieved
```

Comment:
what do you mean by <code>deleteConversation</code> method exactly. Do we have any method like this already? or you mean to create a new method, if yes then what it will do differently. 
And what useless work 'deleteMessages' does ? @azasypkin